ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/OnChainEeaSendRawTransactionTest.java

```diff
@@ -0,0 +1,140 @@
+        .thenReturn(MOCK_ORION_KEY);
+    when(privacyController.validatePrivateTransaction(
+            any(PrivateTransaction.class), any(String.class)))
+        .thenReturn(ValidationResult.valid());
+    final Optional<PrivacyGroup> optionalPrivacyGroup =
+        Optional.of(
+            new PrivacyGroup(
+                "", PrivacyGroup.Type.ONCHAIN, "", "", Arrays.asList(ENCLAVE_PUBLIC_KEY)));
+    when(privacyController.retrieveOnChainPrivacyGroupWithToBeAddedMembers(any(), any(), any()))
+        .thenReturn(optionalPrivacyGroup);
+    when(privacyController.createPrivacyMarkerTransaction(
+            any(String.class), any(PrivateTransaction.class), any(Address.class)))
+        .thenReturn(PUBLIC_TRANSACTION);
+    when(transactionPool.addLocalTransaction(any(Transaction.class)))
+        .thenReturn(ValidationResult.valid());
+
+    final JsonRpcRequestContext request =
+        new JsonRpcRequestContext(
+            new JsonRpcRequest(
+                "2.0",
+                "eea_sendRawTransaction",
+                new String[] {VALID_PRIVATE_TRANSACTION_RLP_PRIVACY_GROUP}));
+
+    final JsonRpcResponse expectedResponse =
+        new JsonRpcSuccessResponse(
+            request.getRequest().getId(),
+            "0x221e930a2c18d91fca4d509eaa3512f3e01fef266f660e32473de67474b36c15");
+
+    final JsonRpcResponse actualResponse = method.response(request);
+
+    assertThat(actualResponse).isEqualToComparingFieldByField(expectedResponse);
+    verify(privacyController).sendTransaction(any(PrivateTransaction.class), any(), any());
+    verify(privacyController)
+        .validatePrivateTransaction(any(PrivateTransaction.class), any(String.class));
+    verify(privacyController)
+        .createPrivacyMarkerTransaction(
+            any(String.class), any(PrivateTransaction.class), eq(Address.ONCHAIN_PRIVACY));
+    verify(transactionPool).addLocalTransaction(any(Transaction.class));
+  }
+
+  @Test
+  public void transactionFailsForLegacyPrivateTransaction() {
+    method =
+        new OnChainEeaSendRawTransaction(
+            transactionPool, privacyController, enclavePublicKeyProvider);
+
+    final JsonRpcRequestContext request = getJsonRpcRequestContext();
+
+    final JsonRpcResponse expectedResponse =
+        new JsonRpcErrorResponse(
+            request.getRequest().getId(), JsonRpcError.ONCHAIN_PRIVACY_GROUP_ID_NOT_AVAILABLE);
+
+    final JsonRpcResponse actualResponse = method.response(request);
+
+    assertThat(actualResponse).isEqualToComparingFieldByField(expectedResponse);
+  }
+
+  private JsonRpcRequestContext getJsonRpcRequestContext() {
+    return new JsonRpcRequestContext(
+        new JsonRpcRequest(
+            "2.0", "eea_sendRawTransaction", new String[] {VALID_LEGACY_PRIVATE_TRANSACTION_RLP}),
+        user);
+  }
+
+  @Test
+  public void offChainPrivacyGroupTransactionFailsWhenOnchainPrivacyGroupFeatureIsEnabled() {
```

Comment:
I'm not sure if this method name is correct. This test seems to be testing what happens when the on-chain privacy group is not found.

Reply:
Already done